lib/commands/file-movement.js

```diff
@@ -1,12 +1,243 @@
 
 
 let commands = iosCommands.file;
 
+async function verifyIFusePresence () {
+  if (!await fs.which('ifuse')) {
+    log.errorAndThrow(`'ifuse' tool is required to be installed on the machine. ` +
+                      `Install it using 'brew cask install osxfuse && brew install ifuse' or check ` +
+                      `if it is available in PATH environment variable if the tool is already installed. ` +
+                      `Current PATH value: ${process.env.PATH}`);
+  }
+}
+
+async function mountDevice (device, iFuseArgs) {
+  log.debug(`Starting ifuse with args '${iFuseArgs}'...`);
+  try {
+    await exec('ifuse', iFuseArgs);
+  } catch (e) {
+    log.errorAndThrow(`Cannot mount the media folder of the device with UDID ${device.udid}. ` +
+                      `Make sure osxfuse plugin has necessary permissions in System Preferences->Security & Privacy. ` +
+                      `Error code: ${e.code}; stderr output: ${e.stderr}`);
+  }
+}
+
+async function parseContainerPath (remotePath, containerRootSupplier) {
+  const match = CONTAINER_PATH_PATTERN.exec(remotePath);
+  if (!match) {
+    log.errorAndThrow(`It is expected that package identifier is separated from the relative path with a single slash. ` +
+                      `'${remotePath}' is given instead`);
+  }
+  const containerRoot = _.isFunction(containerRootSupplier) ?
+    (await containerRootSupplier(match[1])) :
+    containerRootSupplier;
+  return [match[1], path.posix.resolve(containerRoot, match[2])];
+}
+
+/**
+ * Save the given base64 data chunk as a binary file on the Simulator under test.
+ *
+ * @param {Object} device - The device object, which represents the device under test.
+ *                          This object is expected to have the `udid` property containing the
+ *                          valid device ID.
+ * @param {string} remotePath - The remote path on the device. This variable can be prefixed with
+ *                              bundle id, so then the file will be uploaded to the corresponding
+ *                              application container instead of the default media folder, for example
+ *                              '@com.myapp.bla/RelativePathInContainer/111.png'. The '@' character at the
+ *                              beginning of the argument is mandatory in such case.
+ *                              The relative folder path is ignored if the file is going to be uploaded
+ *                              to the default media folder and only the file name is considered important.
+ * @param {string} base64Data - Base-64 encoded content of the file to be uploaded.
+ */
+async function pushFileToSimulator (device, remotePath, base64Data) {
+  if (_.isArray(base64Data)) {
+    // some clients (ahem) java, send a byte array encoding utf8 characters
+    // instead of a string, which would be infinitely better!
+    base64Data = Buffer.from(base64Data).toString('utf8');
+  }
+  const buf = Buffer.from(base64Data, 'base64');
```

Comment:
is Buffer.from supposed to work before Node8? Or Babel takes care of this?

Reply:
yeah, it's supported from node 5